src/js/views/Project/Dependencies.jsx

```diff
@@ -84,17 +91,73 @@ function Dependencies({ project, urlPath }) {
     }
   ]
 
+  const formColumns = [
+    {
+      title: t('terms.projectType'),
+      name: 'id',
+      type: 'number',
+      omitOnAdd: true
+    },
+    {
+      title: t('terms.projectType'),
+      name: 'dependency_id',
+      type: 'project'
+    }
+  ]
+
+  async function onSubmitAdd(formValues) {
+    const url = new URL(
+      `/projects/${project.id}/dependencies`,
+      globalState.baseURL
+    )
+    const result = await httpPost(globalState.fetch, url, formValues)
+    if (result.success === true) {
+      updateDependencies()
+    } else {
+      setErrorMessage(result.data)
+    }
+    setShowForm(false)
```

Comment:
Do you want to leave the form open when it fails so that the user can edit and retry?

Reply:
Good point. I should actually leverage ModalForm's pattern of the onSubmit prop function returning an error on failure, which gets displayed in the modal itself. Currently the error message is on the page, not the modal.